stacker/blueprints/variables/types.py

```diff
@@ -1,15 +1,79 @@
 
 
+class TroposphereType(object):
+
+    def __init__(self, defined_type, resource_name=None):
+        """Represents a Troposphere type.
+
+        :class:`TroposphereType` will convert the value for the variable to the
+        given Troposphere type.
+
+        Args:
+            defined_type (Union[list, type]): List of or single Troposphere
+                type
+            resource_name (str): The name to use for the resource when creating
```

Comment:
I think the way I did it in the security group blueprint actually makes a lot of sense.  You have to give unique names to all resources in a blueprint, so you might as well make it a requirement in the config.  Going to hold off on further review till you reply to this.